Yoast/Yoast-SEO-for-TYPO3
# [FEATURE] Possibility to automatically add Yoast SEO functionality to custom records

## Summary

This PR can be summarized in the following changelog entry:

* Added a possibility to configure full Yoast SEO functionality for custom records by adding a few lines to the TCA Overrides
* For full explanation check the updated documentation
  * Tip: Checkout branch and run `ddev docs` to get the generated version

## Relevant technical choices:
* The setup of `RecordUtility` and `RecordRegistry` is based on the `CategoryRegistry` of core
* The TCA for a registered record is expanded with a `yoast_seo` section with all the settings, in the frontend (and for updating database schema) these sections are read and converted to Record objects which are cached
  * Couldn't find another easy way of doing this, `CategoryRegistry` has its own hardcoded calls within core
* Dropped support for PHP<7.4, this is for the upcoming major version 9

## Test instructions

This PR can be tested by following these steps:

* Following the documentation, configure for a record and see if all the backend fields are available and the frontend renders correctly (pagetitle, metatags, canonical)

## Quality assurance


Fixes #507
